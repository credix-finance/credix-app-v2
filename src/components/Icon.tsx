import React from "react";

type IconNames = "bookmark" | "wallet";

interface IconProps {
	name: IconNames;
	className?: string;
}

export const Icon = ({ name, className }: IconProps) => {
	const defaultClassNames = "w-4 h-4 fill-current";
	className = `${defaultClassNames} ${className ? className : ""}`;

	switch (name) {
		case "bookmark":
			return (
				<svg className={className} viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
					<path d="M2.66699 4.66659C2.66699 2.82564 4.15938 1.33325 6.00033 1.33325H10.0003C11.8412 1.33325 13.3336 2.82564 13.3336 4.66659V12.6594C13.3336 14.3619 11.3421 15.2859 10.0421 14.1865L8.43082 12.8238C8.1823 12.6136 7.81834 12.6136 7.56982 12.8238L5.95848 14.1865C4.65855 15.2859 2.66699 14.3619 2.66699 12.6594V4.66659ZM6.00033 2.66659C4.89576 2.66659 4.00033 3.56202 4.00033 4.66659V12.6594C4.00033 13.2269 4.66418 13.5349 5.09749 13.1684L6.70883 11.8057C7.45439 11.1752 8.54626 11.1752 9.29182 11.8057L10.9031 13.1684C11.3364 13.5349 12.0003 13.2269 12.0003 12.6594V4.66659C12.0003 3.56202 11.1049 2.66659 10.0003 2.66659H6.00033Z" />
				</svg>
			);
		case "wallet":
			return (
				<svg className={className} viewBox="0 0 24 25" xmlns="http://www.w3.org/2000/svg">
					<path d="M17 9.5h5v6h-5a3 3 0 1 1 0-6Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z" />
					<path d="M7.759 4.5h8.482c.805 0 1.47 0 2.01.044.563.046 1.08.145 1.565.392a4 4 0 0 1 1.748 1.748c.247.485.346 1.002.392 1.564.044.541.044 1.206.044 2.01v4.483c0 .805 0 1.47-.044 2.01-.046.563-.145 1.08-.392 1.565a4 4 0 0 1-1.748 1.748c-.485.247-1.002.346-1.564.392-.541.044-1.206.044-2.01.044H7.758c-.805 0-1.47 0-2.01-.044-.563-.046-1.08-.145-1.565-.392a4 4 0 0 1-1.748-1.748c-.247-.485-.346-1.002-.392-1.564C2 16.21 2 15.546 2 14.742v-4.483c0-.805 0-1.47.044-2.01.046-.563.145-1.08.392-1.565a4 4 0 0 1 1.748-1.748c.485-.247 1.002-.346 1.564-.392C6.29 4.5 6.954 4.5 7.758 4.5ZM5.91 6.538c-.438.035-.663.1-.819.18a2 2 0 0 0-.874.874c-.08.156-.145.38-.18.819C4 8.861 4 9.443 4 10.3v4.4c0 .857 0 1.439.038 1.889.035.438.1.663.18.819a2 2 0 0 0 .874.874c.156.08.38.145.819.18.45.037 1.032.038 1.889.038h8.4c.857 0 1.439 0 1.889-.038.438-.035.663-.1.819-.18a2 2 0 0 0 .874-.874c.08-.156.145-.38.18-.819.037-.45.038-1.032.038-1.889v-4.4c0-.857 0-1.439-.038-1.889-.035-.438-.1-.663-.18-.819a2 2 0 0 0-.874-.874c-.156-.08-.38-.145-.819-.18-.45-.037-1.032-.038-1.889-.038H7.8c-.857 0-1.439 0-1.889.038Z" />
				</svg>
			);
		default:
			return null;
	}
};
