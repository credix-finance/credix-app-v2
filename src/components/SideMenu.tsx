import { useRouter } from "next/router"
import React from "react";
import { CredixLogo } from "./CredixLogo"
import { SideMenuItem, SideMenuItemProps } from "./SideMenuItem";

interface SideMenuProps {
	menuItems: Omit<SideMenuItemProps, "isActive">[];
}

export const SideMenu = ({ menuItems }: SideMenuProps) => {
	const router = useRouter();

	return (
		<div className="w-72 bg-darker h-screen pt-[33px] sticky top-0">
			<div className="w-[39px] h-[39px] text-credix-primary ml-[43px]">
				<CredixLogo mode="light" />
			</div>
			<div className="mt-[145px]">
				{menuItems.map((item) => (
					<SideMenuItem key={item.label} isActive={router.pathname === item.path} {...item} />
				))}
			</div>
		</div>
	);
};
