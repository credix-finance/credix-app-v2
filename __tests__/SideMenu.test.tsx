import React from "react";
import renderer from "react-test-renderer";
import { SideMenu } from "@components/SideMenu";
import { IconName } from "@components/Icon";

const items = [
	{
		icon: "line-chart" as IconName,
		label: "Invest",
		path: "/invest",
		isActive: true,
	},
	{
		icon: "coins" as IconName,
		label: "withdraw",
		path: "/withdraw",
		isActive: false,
	},
	{
		icon: "timeline" as IconName,
		label: "Deals",
		path: "/deals",
		isActive: false,
	},
	{
		icon: "grid" as IconName,
		label: "Transactions",
		path: "/transactions",
		isActive: false,
	},
	{
		icon: "user" as IconName,
		label: "Account",
		path: "/account",
		isActive: false,
	},
];

test("Side menu", () => {
	const component = renderer.create(<SideMenu menuItems={items} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
