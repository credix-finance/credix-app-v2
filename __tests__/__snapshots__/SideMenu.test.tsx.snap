// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side menu 1`] = `
<div
  className="w-72 bg-darker h-screen pt-[33px] sticky top-0"
>
  <div
    className="text-credix-primary ml-[43px]"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <svg
        className="w-7 h-10 text-white "
        data-name="Layer 1"
        fill="currentColor"
        id="Layer_1"
        viewBox="187.95 93.98 59.68 82.96"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M188,105.28c0-7.23,2.66-11.3,8.19-11.3,4.23,0,6.17,2.44,7.17,5.41l-3.16,1.07c-.66-2-1.56-3.57-4-3.57-3.09,0-4.44,2.53-4.44,6.54v3.69c0,4,1.35,6.54,4.44,6.54,2.48,0,3.38-1.56,4-3.56l3.16,1.06c-1,3-2.94,5.41-7.17,5.41C190.61,116.57,188,112.51,188,105.28Z"
        />
        <path
          d="M214.15,116.2h-3.48V94.36h8.17c4.13,0,6.41,2.5,6.41,6.57,0,3.31-1.68,5.63-4.72,6.13l5,9.14h-3.88l-4.64-8.83h-2.9Zm4.19-11.64c2.19,0,3.25-1,3.25-3v-1.34c0-2-1.06-3-3.25-3h-4.19v7.29Z"
        />
        <path
          d="M232.65,116.2V94.36h13.67V97.3H236.15v6.35H246v2.94h-9.8v6.67h10.17v2.94Z"
        />
        <path
          d="M188,129.34h6.51c5.53,0,8.23,3.82,8.23,10.92s-2.7,10.93-8.23,10.93H188Zm6.29,19c3.19,0,4.75-2.26,4.75-6.23v-3.63c0-4-1.56-6.23-4.75-6.23h-2.82v16.09Z"
        />
        <path
          d="M209.79,151.19v-2.66h5.14V132h-5.14v-2.66h13.77V132h-5.13v16.53h5.13v2.66Z"
        />
        <path
          d="M247.42,151.19h-4L241,146.87l-2.32-4.13h-.18l-2.38,4.13-2.57,4.32H229.8L236.49,140l-6.28-10.64h4l2.22,3.91,2.25,4h.16l2.25-4,2.32-3.91h3.72L240.78,140Z"
        />
        <rect
          height="3"
          width="59.68"
          x="187.95"
          y="173.94"
        />
      </svg>
    </a>
  </div>
  <div
    className="mt-[145px]"
  >
    <a
      className="h-[60px] flex items-center bg-credix-primary group hover:bg-neutral-10"
      href="/invest"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="
						border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center
						group-hover:border-neutral-10 group-hover:text-darker group-hover:text-bold
						text-darker font-bold
					"
      >
        <svg
          className="w-4 h-4 fill-current w-[20px] h-[20px]"
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="m14.034 7.569-1.225 2.45a2.493 2.493 0 0 1 1.49.745l1.226-2.45a2.494 2.494 0 0 1-1.491-.745Zm-2.612 2.675a2.51 2.51 0 0 0-1.178 1.178L8.578 9.756a2.51 2.51 0 0 0 1.178-1.178l1.666 1.666ZM5.701 9.235l-1.225 2.45a2.493 2.493 0 0 1 1.49.746l1.225-2.45a2.493 2.493 0 0 1-1.49-.746Z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M16.667 5.833a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-10 1.667a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0ZM10 7.5a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-5 6.667a.833.833 0 1 1-1.666 0 .833.833 0 0 1 1.666 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm5.833-.834a.833.833 0 1 0 0-1.666.833.833 0 0 0 0 1.666Zm0 1.667a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pl-[18px] text-base"
        >
          Invest
        </span>
      </div>
    </a>
    <a
      className="h-[60px] flex items-center bg-darker group hover:bg-neutral-10"
      href="/withdraw"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="
						border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center
						group-hover:border-neutral-10 group-hover:text-darker group-hover:text-bold
						text-credix-primary font-medium
					"
      >
        <svg
          className="w-4 h-4 fill-current w-[20px] h-[20px]"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.167 5a2.5 2.5 0 0 1 2.5-2.5H10a2.5 2.5 0 0 1 .658 4.912c.113.285.175.596.175.921 0 .446-.116.864-.32 1.226.159.102.306.222.437.356a4.19 4.19 0 0 0-1.15 1.21.831.831 0 0 0-.633-.292H5.833a.833.833 0 0 0 0 1.667h3.334a.837.837 0 0 0 .084-.004 4.155 4.155 0 0 0 1.195 3.842A2.498 2.498 0 0 1 8.333 17.5H5a2.5 2.5 0 0 1-1.346-4.607 2.489 2.489 0 0 1-.32-1.226c0-.446.116-.864.32-1.227a2.501 2.501 0 0 1 .688-4.52A2.493 2.493 0 0 1 4.167 5Zm4.166 2.5a.833.833 0 1 1 0 1.667H5A.833.833 0 1 1 5 7.5h3.333ZM10 5.833H6.667a.833.833 0 1 1 0-1.666H10a.833.833 0 0 1 0 1.666Zm-1.667 8.334a.833.833 0 1 1 0 1.666H5a.833.833 0 1 1 0-1.666h3.333Z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M13.334 10.833a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5Zm-4.167 2.5a4.167 4.167 0 1 1 8.333 0 4.167 4.167 0 0 1-8.333 0Z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pl-[18px] text-base"
        >
          withdraw
        </span>
      </div>
    </a>
    <a
      className="h-[60px] flex items-center bg-darker group hover:bg-neutral-10"
      href="/deals"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="
						border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center
						group-hover:border-neutral-10 group-hover:text-darker group-hover:text-bold
						text-credix-primary font-medium
					"
      >
        <svg
          className="w-4 h-4 fill-current w-[20px] h-[20px]"
          viewBox="0 0 20 21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M6.666 6.032a.833.833 0 1 0 0 1.666H10a.833.833 0 0 0 0-1.666H6.666Zm.834 4.166c0-.46.373-.833.833-.833h3.333a.833.833 0 1 1 0 1.667H8.333a.833.833 0 0 1-.833-.834Zm1.666 3.334c0-.46.373-.834.834-.834h3.333a.833.833 0 1 1 0 1.667H10a.833.833 0 0 1-.834-.833Z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M8.631 2.698h2.738c.903 0 1.632 0 2.222.048.607.05 1.14.155 1.634.406a4.167 4.167 0 0 1 1.82 1.821c.252.494.357 1.027.407 1.635.048.59.048 1.318.048 2.221v2.738c0 .903 0 1.632-.048 2.222-.05.607-.155 1.14-.406 1.634a4.167 4.167 0 0 1-1.821 1.821c-.494.252-1.027.356-1.634.406-.59.048-1.319.048-2.222.048H8.631c-.903 0-1.632 0-2.222-.048-.607-.05-1.14-.154-1.634-.406a4.167 4.167 0 0 1-1.82-1.82c-.252-.494-.357-1.028-.407-1.635-.048-.59-.048-1.319-.048-2.222V8.83c0-.903 0-1.631.048-2.221.05-.608.155-1.141.406-1.635a4.167 4.167 0 0 1 1.821-1.82c.494-.252 1.027-.357 1.634-.407.59-.048 1.319-.048 2.222-.048Zm-2.086 1.71c-.504.04-.794.118-1.013.23A2.5 2.5 0 0 0 4.439 5.73c-.112.22-.188.51-.23 1.013-.042.514-.042 1.175-.042 2.122v2.667c0 .947 0 1.607.042 2.121.042.504.118.794.23 1.014a2.5 2.5 0 0 0 1.093 1.092c.22.112.509.189 1.013.23.514.042 1.175.043 2.122.043h2.666c.947 0 1.608-.001 2.122-.043.504-.041.794-.118 1.013-.23a2.5 2.5 0 0 0 1.093-1.092c.112-.22.188-.51.23-1.014.042-.514.042-1.174.042-2.121V8.865c0-.947 0-1.608-.042-2.122-.042-.504-.118-.794-.23-1.013a2.5 2.5 0 0 0-1.093-1.093c-.22-.111-.509-.188-1.013-.23-.514-.041-1.175-.042-2.122-.042H8.667c-.947 0-1.608 0-2.122.043Z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pl-[18px] text-base"
        >
          Deals
        </span>
      </div>
    </a>
    <a
      className="h-[60px] flex items-center bg-darker group hover:bg-neutral-10"
      href="/transactions"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="
						border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center
						group-hover:border-neutral-10 group-hover:text-darker group-hover:text-bold
						text-credix-primary font-medium
					"
      >
        <svg
          className="w-4 h-4 fill-current w-[20px] h-[20px]"
          viewBox="0 0 20 21"
          xmlns="http://www.ww3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M14.193 2.77h-.053c-.366 0-.68 0-.942.018-.275.018-.55.06-.821.172a2.5 2.5 0 0 0-1.353 1.353 2.559 2.559 0 0 0-.173.822c-.018.26-.018.576-.018.942v1.72c0 .365 0 .68.018.942.019.274.06.55.173.821a2.5 2.5 0 0 0 1.353 1.353c.271.112.546.154.821.172.261.018.576.018.942.018h.053c.366 0 .681 0 .942-.018.275-.018.55-.06.822-.172A2.5 2.5 0 0 0 17.31 9.56c.112-.272.154-.547.172-.821.018-.261.018-.577.018-.942v-1.72c0-.366 0-.681-.018-.942a2.556 2.556 0 0 0-.172-.822 2.5 2.5 0 0 0-1.353-1.353 2.557 2.557 0 0 0-.822-.172c-.26-.018-.576-.018-.942-.018ZM13.014 4.5a.968.968 0 0 1 .298-.05c.197-.013.455-.013.855-.013.4 0 .657 0 .855.014.19.012.262.035.297.049a.833.833 0 0 1 .45.451.967.967 0 0 1 .05.297c.014.198.014.456.014.855V7.77c0 .4 0 .658-.014.855a.967.967 0 0 1-.05.297.833.833 0 0 1-.45.451.966.966 0 0 1-.297.05c-.198.013-.456.014-.855.014-.4 0-.658 0-.855-.014a.967.967 0 0 1-.298-.05.833.833 0 0 1-.45-.45.97.97 0 0 1-.05-.298 14.16 14.16 0 0 1-.014-.855V6.103c0-.4 0-.657.014-.855a.97.97 0 0 1 .05-.297.833.833 0 0 1 .45-.451ZM5.86 9.437h-.053c-.366 0-.681 0-.942.017-.275.02-.55.06-.822.173A2.5 2.5 0 0 0 2.69 10.98a2.558 2.558 0 0 0-.172.821c-.018.261-.018.577-.018.942v1.72c0 .366 0 .681.018.942.019.275.06.55.172.822a2.5 2.5 0 0 0 1.353 1.353c.272.112.547.153.822.172.26.018.576.018.942.018h.053c.366 0 .68 0 .942-.018.275-.019.55-.06.821-.172a2.5 2.5 0 0 0 1.353-1.353c.113-.272.154-.547.173-.822.018-.26.018-.576.018-.942v-1.72c0-.365 0-.68-.018-.942a2.558 2.558 0 0 0-.173-.821 2.5 2.5 0 0 0-1.353-1.353 2.557 2.557 0 0 0-.821-.173c-.261-.017-.576-.017-.942-.017Zm-1.179 1.73a.967.967 0 0 1 .297-.05c.198-.013.456-.014.855-.014.4 0 .658 0 .856.014.19.013.262.035.297.05a.833.833 0 0 1 .45.45.967.967 0 0 1 .05.298c.014.197.014.455.014.855v1.667c0 .4 0 .657-.014.855a.966.966 0 0 1-.05.297.833.833 0 0 1-.45.45.963.963 0 0 1-.297.05c-.198.014-.456.014-.856.014-.4 0-.657 0-.855-.014a.963.963 0 0 1-.297-.05.833.833 0 0 1-.45-.45.966.966 0 0 1-.05-.297 14.098 14.098 0 0 1-.014-.855V12.77c0-.4 0-.658.014-.855a.967.967 0 0 1 .05-.297.833.833 0 0 1 .45-.451ZM5 2.77a2.5 2.5 0 0 0 0 5h1.667a2.5 2.5 0 0 0 0-5H5Zm-.833 2.5c0-.46.373-.833.833-.833h1.667a.833.833 0 0 1 0 1.666H5a.833.833 0 0 1-.833-.833ZM13.333 12.77a2.5 2.5 0 0 0 0 5H15a2.5 2.5 0 0 0 0-5h-1.667Zm-.833 2.5c0-.46.373-.833.833-.833H15a.833.833 0 0 1 0 1.666h-1.667a.833.833 0 0 1-.833-.833Z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pl-[18px] text-base"
        >
          Transactions
        </span>
      </div>
    </a>
    <a
      className="h-[60px] flex items-center bg-darker group hover:bg-neutral-10"
      href="/account"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="
						border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center
						group-hover:border-neutral-10 group-hover:text-darker group-hover:text-bold
						text-credix-primary font-medium
					"
      >
        <svg
          className="w-4 h-4 fill-current w-[20px] h-[20px]"
          viewBox="0 0 20 21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.5 13.603a2.5 2.5 0 0 0-2.5 2.5c0 .46.373.834.833.834h8.333c.46 0 .834-.373.834-.834a2.5 2.5 0 0 0-2.5-2.5h-5Zm-4.167 2.5A4.167 4.167 0 0 1 7.5 11.937h5a4.167 4.167 0 0 1 4.166 4.166 2.5 2.5 0 0 1-2.5 2.5H5.833a2.5 2.5 0 0 1-2.5-2.5ZM10 3.603a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5Zm-4.167 2.5a4.167 4.167 0 1 1 8.333 0 4.167 4.167 0 0 1-8.333 0Z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="pl-[18px] text-base"
        >
          Account
        </span>
      </div>
    </a>
  </div>
</div>
`;
